test/angular-test-init-spec.ts

```diff
@@ -0,0 +1,59 @@
+import {TestBed} from '@angular/core/testing';
+import {
+  BrowserDynamicTestingModule,
+  platformBrowserDynamicTesting,
+} from '@angular/platform-browser-dynamic/testing';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+
+
+/* Common setup / initialization for all unit tests in Angular Material and CDK. */
+
+
+const testBed = TestBed.initTestEnvironment(
+    [BrowserDynamicTestingModule, NoopAnimationsModule], platformBrowserDynamicTesting());
+patchTestBedToDestroyFixturesAfterEveryTest(testBed);
+
+(window as any).module = {};
+(window as any).isNode = false;
+(window as any).isBrowser = true;
+(window as any).global = window;
+
+
+/**
+ * Monkey-patches TestBed.resetTestingModule such that any errors that occur during component
+ * destruction are thrown instead of silently logged. Also runs TestBed.resetTestingModule after
+ * each unit test.
+ *
+ * Without this patch, the combination of two behaviors is problematic for Angular Material:
+ * - TestBed.resetTestingModule catches errors thrown on fixture destruction and logs them without
+ *     the errors ever being thrown. This means that any component errors that occur in ngOnDestroy
+ *     can encounter errors silently and still pass unit tests.
+ * - TestBed.resetTestingModule is only called *before* a test is run, meaning that even *if* the
+ *    aforementioned errors were thrown, they would be reported for the wrong test (the test that's
+ *    about to start, not the test that just finished).
+ */
+function patchTestBedToDestroyFixturesAfterEveryTest(testBed) {
+  // Original resetTestingModule function of the TestBed.
+  const _resetTestingModule = testBed.resetTestingModule;
+
+  // Monkey-patch the resetTestingModule to destroy fixtures outside of a try/catch block.
+  // With https://github.com/angular/angular/commit/2c5a67134198a090a24f6671dcdb7b102fea6eba
+  // errors when destroying components are no longer causing Jasmine to fail.
+  testBed.resetTestingModule = function() {
```

Comment:
are fat-arrows not allowed in this file?

Reply:
It was originally a JS file that I copy-pasted. Fixed.

(the one `function` remaining is necessary since it deal with `this`)